hub/api/tests/test_api.py

```diff
@@ -0,0 +1,75 @@
+import os, shutil
+import numpy as np
+from hub.api.dataset import Dataset
+from hub.core.chunk_engine.read import read_dataset_meta, read_tensor_meta
+import pytest
+
+
+@pytest.fixture(autouse=True)
+def clear_test_files():
+    shutil.rmtree("/tmp/hub-test/", ignore_errors=True)
+
+
+def test_create_empty_memory_dataset():
+    ds = Dataset("username/data")
+    assert ds.slice == slice(None)
+    assert read_dataset_meta(ds.provider) == {"tensors": []}
+
+
+def test_create_empty_local_dataset():
+    ds = Dataset("/tmp/hub-test/empty")
+    assert read_dataset_meta(ds.provider) == {"tensors": []}
+
+
+def test_populate_local_dataset():
+    ds = Dataset("/tmp/hub-test/populate")
```

Comment:
can you add a constant at the top of this file that contains "/tmp/hub-test/"?

Reply:
also, could you add a TODO here for "create pytest Dataset fixture"? i would love for these Datasets to be created implicitly with the storage provider & all and you just write a test with "ds" as a parameter

Reply:
Done!